modernize release notes generator

abbra opened the pull request:
* Add support for RN: markup in the commit messages
* Refresh the release notes template
* Use major version in the template automatically

abbra committed:
Protect against github API errors

abbra committed:
Make sure only run backports when there are backports

abbra committed:
Add support for release notes marks in the commit messages

Commit messages can contain release notes with the follow format:

RN: first line
RN: second line
RN: ...
RN: end of release note

The release note will be attached to the changelog field in any ticket
associated with the commit.

abbra committed:
Modernize wiki template

Remove mentioning of COPR repositories, we deliver to Fedora directly.

Use major version in the template automatically.

abbra commented:
An example of new formatting can be found in this email: https://lists.fedorahosted.org/archives/list/freeipa-devel@lists.fedorahosted.org/thread/OEJGO6YP3LLAKHUN5XICBR7EGGHNQC7F/

rcritten commented on ipatool:
```diff
@@ -928,7 +928,10 @@ def backport(ctx, backport_branches, repo, pr):
-    github_login = github3.login(token=ctx.config['gh-token']).me().login
+    try:
+        github_login = github3.login(token=ctx.config['gh-token']).me().login
+    except KeyError as exc:
```
unused variable exc, should the exception be at least printed?

abbra replied:
Will do.

rcritten commented on ipatool:
```diff
@@ -1140,7 +1143,8 @@ def pr_push_command(ctx):
                 backport_branches.update(l for l in labels if pat.match(l))
-            backport(ctx, backport_branches, repo, pr)
+            if backport_branches:
+                backport(ctx, backport_branches, repo, pr)
```
Is this change related the notes generator?

abbra replied:
No, this was part of my tree for some time because of the failures I saw occasionally with github. Had to put it somewhere.

abbra committed:
Add support for known issues

In Pagure issues there is a field 'knownissue' ('true'/'false') that can
be used to specify whether this issue currently has no solution. A
changelog entry for the issue (either from commit message as a release
note or from the ticket 'changelog' field) can be used to document the
issue for release notes.

The note will be added to 'Known issues' section.

abbra committed:
Allow changelog without milestone tickets

For minor releases it is often preferable to remove primary milestone
tickets. Instead of specifying incorrect milestone, add --nomilestones
option that avoids querying Pagure at all.

rcritten commented on release/release-notes.py:
```diff
@@ -297,6 +330,9 @@ def parse_args():
                         metavar='GIT_DIR', default=GIT_DIR)
+    parser.add_argument('--nomilestones', dest='nomilestones',
+                        action="store_true",
+                        help="Only use tickets mentioned in the commmits")
```
nit: commmits -> commits

rcritten commented:
Just one last request: can you add one or more sample invocations somewhere? There is currently one in release-notes.py but I would also be fine with the addition of a README.md where things like RN can be covered as well.

abbra committed:
Add README for the release notes tool

rcritten commented:
Looks great to me, ack.

abbra committed:
Add release notes explanation

abbra commented:
I added a bit more details about release notes gathering.

rcritten commented:
Still LGTM.
